sbi/utils/sbiutils.py

```diff
@@ -31,20 +32,59 @@ def clamp_and_warn(name: str, value: float, min_val: float, max_val: float) -> f
     return clamped_val
 
 
-class Standardize(nn.Module):
+def standardizing_transform(
+    batch_t: Tensor, min_std: float = 1e-7
+) -> transforms.AffineTransform:
+    """Builds standardizing transform
+
+    Args:
+        batch_t: Batched tensor from which mean and std deviation (across 
+            first dimension) are computed.
+        min_std:  Minimum value of the standard deviation to use when z-scoring to
+            avoid division by zero.
+
+    Returns:
+        Affine transform for z-scoring
     """
-    Standardize inputs, i.e. subtract mean and divide by standard deviation. Inherits
-    from `nn.Module`, hence it can be used as a step in `nn.Sequential`.
+
+    is_valid_t, *_ = handle_invalid_x(batch_t, True)
+
+    t_mean = torch.mean(batch_t[is_valid_t], dim=0)
+    t_std = torch.std(batch_t[is_valid_t], dim=0)
+    t_std[t_std < min_std] = min_std
+
+    return transforms.AffineTransform(shift=-t_mean / t_std, scale=1 / t_std)
+
+
+def standardizing_net(batch_t: Tensor, min_std: float = 1e-7) -> nn.Module:
+    """Builds standardizing network
+
+    Args:
+        batch_t: Batched tensor from which mean and std deviation (across 
+            first dimension) are computed.
+        min_std:  Minimum value of the standard deviation to use when z-scoring to
+            avoid division by zero.
+
+    Returns:
+        Neural network module for z-scoring
     """
 
-    def __init__(self, mean, std):
-        super(Standardize, self).__init__()
-        self.mean = mean
-        self.std = std
+    class Standardize(nn.Module):
+        def __init__(self, mean, std):
+            super(Standardize, self).__init__()
+            self.mean = mean
+            self.std = std
+
+        def forward(self, tensor):
```

Comment:
No guard against division-by-zero needed here?

Reply:
No, the `min_std` is take care of outside of the module